### PR TITLE
Show `index_errors` in docs

### PR DESCRIPTION
**Pull request by GarrisonD**
### Motivation / Background

This Pull Request has been created because there is no info about `index_errors` in docs now.

### Detail

This Pull Request adds missing `+`s for the `index_errors` section.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by GarrisonD** [comment]
Also if this MR is going to be approved, we need to cherry-pick the changes to the docs of Rails 5.x and 6.x

File: activerecord/lib/active_record/associations.rb [hunk 1]
```diff
@@ -1244,7 +1244,7 @@ module ClassMethods
         #   Serves as a composite foreign key. Defines the list of columns to be used to query the associated object.
         #   This is an optional option. By default Rails will attempt to derive the value automatically.
         #   When the value is set the Array size must match associated model's primary key or +query_constraints+ size.
-        # [:index_errors]
+        # [+:index_errors]
```
Comment (skipkayhil) [inline activerecord/lib/active_record/associations.rb:1247]:
```suggestion
        # [+:index_errors+]
```

Checkout the `docs-preview` job in CI, you can see what the change will look like when published:

image

Comment (GarrisonD) [reply activerecord/lib/active_record/associations.rb:1247]: You are right, didn't pay enough attention to that.

**Review by skipkayhil — APPROVED** [review]
image

:+1: Looks good!

**Comment by skipkayhil** [comment]
Backported to `7-2-stable` in 43530b4

I also backported the `index_errors` doc (pre #48727) to `7-1-stable` in e41965358f308a04ba3f10a184216386485d5403

All other branches are security release only or EOL, so I can't backport any further than that.

**Pull request merged**